src/components/page/page.comp.tsx

```diff
@@ -144,8 +143,14 @@ const PageComp = ({ context }: IProps) => {
         throw new Error('Extracted data is invalid.');
       }
 
+      if (typeof dataTransform === 'function') {
+        for (let item of extractedData) {
```

Comment:
Maybe instead of passing a single item at a time, we should pass the whole data array?
This way you'll be able to add more logic in a higher level.

Reply:
Good idea! done.